### DISCUSSION TITLE
Suggestion to rename field in the `Broker` struct

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/evm-monorepo#325] [General]
The `Broker` struct has two fields that provide:

1. The address to send those funds to.
2. The amount of fees requested by the broker.

https://github.com/sablierhq/v2-core/blob/b23b0654e71849b2a6fd30181e3621b4bb81842d/src/types/DataTypes.sol#L11-L14

I think `addr` here is the only abbreviated variable name we use in the contracts. Isn't it better to rename it to `address` as that's a reserved keyword. But can we maybe go with `Broker.target` or `Broker.account`?

**Comment by andreivladbrg** [comment]
> the amount of fees requested by the broker

That's the percentage not the "amount".

> Broker.target

I don't like it because the word "target" is usually used for proxies.

> Broker.account 

It would work for me.

**Reply by razgraf** [reply]
Yet another good point: maybe "fee" should be renamed to `feePercentage` for clarity?

**Reply by andreivladbrg** [reply]
Not really, all the amount arguments have that keyword. Ex: `totalAmount`

**Reply by PaulRBerg** [reply]
> maybe "fee" should be renamed to `feePercentage` for clarity?

The fact that the `broker.fee` is a fixed-point value is implied by the use of the `UD60x18` type - speaking of which, I invite you and @gavriliumircea to skim through PRBMath's [README](https://github.com/PaulRBerg/prb-math), since Sablier V2 borrows a few core concepts from there.

I have purposefully stayed away from any reference to percentages due to the risk of accidental misuse. That is, we would then have to add warnings surrounding what 100% means - does it mean 1e18 or 1e20? Of course, 100% is 1e18, but by naming this variable `feePercentage`, we might give the false impression that 100% means 1e20 instead.

At any rate, I wouldn't worry too much about naming here. As I said in my comment below, the broker struct is something that only developers will interact with. We will have documentation that explains how to use this struct, and the fact that the `fee` is a percentage fixed-point with up to 18 decimals.
 
> all the amount arguments have that keyword. Ex: `totalAmount`

@andreivladbrg - I did not quite understand what you meant by your comment. Might you have misunderstood @razgraf's point?

**Reply by razgraf** [reply]
I take back the point on fees, it was a secondary thought anyway. It's fine to keep it this way as with Sablier all fees are percentage based. That's a good heuristic we can use in our docs or when explaining the API of the protocol.

**Reply by PaulRBerg** [reply]
Great. Also, FWIW, we have a clear distinction between non-amount and amount variables - the latter either end with the `Amount` suffix, or are encapsulated in a struct that ends with `Amounts` (e.g. [`CreateAmounts`](https://github.com/sablierhq/v2-core/blob/b23b0654e71849b2a6fd30181e3621b4bb81842d/src/types/DataTypes.sol#L33-L37)).

**Reply by andreivladbrg** [reply]
> @andreivladbrg - I did not quite understand what you meant by your comment. Might you have misunderstood @razgraf's point?

I understood his point. 

What I mean by that comment is what you said here: "the latter either end with the Amount suffix, or are encapsulated in a struct that ends with Amounts (e.g. CreateAmounts)."

**Comment by PaulRBerg** [comment]
> The amount of fees requested by the broker

As Andrei pointed out, that is the percentage fee, as indicated by the `UD60x18` type. Amounts are never fixed-point because they can have a variable number of decimals (different ERC-20 tokens have different number of `decimals`).

> I think `addr` here is the only abbreviated variable name

Yeah, but it's fine in this context:

- The `Broker` struct is meant to be seen and used by developers only.
- `addr` is a common abbreviation in the Solidity ecosystem (see, for instance, the [`addr`](https://github.com/foundry-rs/forge-std/blob/662ae0d6936654c5d1fb79fc15f521de28edb60e/src/Vm.sol#L39) cheatcode in Foundry).

> can we maybe go with `Broker.target` or `Broker.account`?

Just like Andrei, I don't like `target`. Not because of the potential overlap with the proxy contracts, but because it's simply not a good fit in this context.

However, I like `account`. I have thought about it myself when I first wrote the struct, but I ended up going with `addr` because the broker address could also be a smart contract address and "account" is typically associated with EOAs. However, this convention has sort of been broken by the ERC-20 standard, which uses the word "account" in the param definition of the `balanceOf` function. So we might just want to go with `account` over `addr` due to the added clarity.

**Comment by PaulRBerg** [comment]
Locking since we have agreed upon using `broker.account`, and I implemented this change in https://github.com/sablierhq/v2-core/pull/334.
